lib/serializer-compiler.js

```diff
@@ -5,8 +5,8 @@ const AjvJTD = require('ajv/dist/jtd')
 const defaultAjvOptions = require('./default-ajv-options')
 
 class SerializerCompiler {
-  constructor (externalSchemas, options) {
-    this.ajv = new AjvJTD(Object.assign({}, defaultAjvOptions, options))
+  constructor (externalSchemas, options = {}) {
+    this.ajv = new AjvJTD(Object.assign({}, defaultAjvOptions, options.customOptions))
```

Comment:
@Eomm 
I think this never worked.

Reply:
```suggestion
    this.ajv = new AjvJTD(Object.assign({}, defaultAjvOptions, options))
```

(true what @climba03003 said)

Let me add more info:

You can use JTD as serializer and/or validator

The `customOptions` comes from `fastifyOpts.ajv.customOptions` that is forwarded to the validation

ANY serializer should get `fastifyOpts.serializerOpts` (that is only 2 levels)

Since all tests are green, I think we should add a test using the `allError: true` parameter (I can do it in a follow up PR)
https://ajv.js.org/guide/getting-started.html#parsing-and-serializing-json

I think that, when I wrote this, the AJV's JTD serializer did not accept any options.

Reply:
As improvement: we could rename/refactor the fastify `ajv` parameter, but, at that time, we wanted to be clear to the user that it was an "ajv" stuff under the hood